Add language support for c++14.

shahamran opened the pull request:
Add a [specification file](https://cms.readthedocs.io/en/v1.3/Configuring%20a%20contest.html#programming-languages) for the C++ 14 standard.
The flag `std=c++14` is valid in `g++` [from version `5.2` and on](https://gcc.gnu.org/onlinedocs/gcc-5.2.0/gcc/C-Dialect-Options.html#C-Dialect-Options) (the ioi server uses a more recent version).
The added file is a copy of to the `c++11` file, with the relevant changes to the standard version.

shahamran committed:
Add language support for c++14.

nir-lavee commented:
Thanks!
The latest [IOI environment](https://ioi2018.jp/competition/competition-environment/) specifies `std=gnu++14` rather than `std=c++14`. Do you know what the differences are? I don't, but I suggest we conform just to be safe. Feel free to change and then merge :)

shahamran committed:
Update c++14_g++.py

Change the `-std=c++14` to `-std=gnu++14`, which enables the [GNU compiler C++ extensions](https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Extensions.html).

shahamran commented:
According to [SO](https://stackoverflow.com/questions/10613126/what-are-the-differences-between-std-c11-and-std-gnu11), the difference is whether or not the [GCC extensions](https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Extensions.html#C_002b_002b-Extensions) are enabled.
I changed the compilation flag. Thanks!
